src/ucs/sys/iovec.inl

```diff
@@ -0,0 +1,129 @@
+                          _src_iov, _src_iov_cnt, \
+                          _src_iov_get_buffer_f, _src_iov_get_length_f, \
+                          _max_length, _iov_iter_p) \
+   ({ \
+        size_t __remain_length = _max_length; \
+        size_t __dst_iov_it    = 0; \
+        size_t __src_iov_it    = (_iov_iter_p)->iov_offset; \
+        size_t __dst_iov_length, __calc_iov_length; \
+        \
+        while ((__src_iov_it < (_src_iov_cnt)) && (__remain_length != 0) && \
+               (__dst_iov_it < *(_dst_iov_cnt_p))) { \
+            ucs_assert(_src_iov_get_length_f(&(_src_iov)[__src_iov_it]) >= \
+                       (_iov_iter_p)->buffer_offset); \
+            __calc_iov_length = _src_iov_get_length_f(&(_src_iov)[__src_iov_it]) - \
+                                (_iov_iter_p)->buffer_offset; \
+            if (__calc_iov_length == 0) { \
+                /* Avoid zero length elements in resulted IOV */ \
+                ++__src_iov_it; \
+                continue; \
+            } \
+            \
+            __dst_iov_length = ucs_min(__calc_iov_length, __remain_length); \
+            \
+            _dst_iov_set_length_f(&(_dst_iov)[__dst_iov_it], __dst_iov_length); \
+            _dst_iov_set_buffer_f(&(_dst_iov)[__dst_iov_it], \
+                                  UCS_PTR_BYTE_OFFSET(_src_iov_get_buffer_f( \
+                                                          &(_src_iov)[__src_iov_it]), \
+                                                      (_iov_iter_p)->buffer_offset)); \
+            \
+            if (__calc_iov_length > __remain_length) { \
+                (_iov_iter_p)->buffer_offset += __remain_length; \
+            } else { \
+                ucs_assert(((_iov_iter_p)->buffer_offset == 0) || \
+                           (__src_iov_it == (_iov_iter_p)->iov_offset)); \
+                (_iov_iter_p)->buffer_offset = 0; \
+                ++__src_iov_it; \
+            } \
+            \
+            ucs_assert(__remain_length >= __dst_iov_length); \
+            __remain_length            -= __dst_iov_length; \
+            ++__dst_iov_it; \
+            \
+        } \
+        \
+        ucs_assert(__dst_iov_it <= *(_dst_iov_cnt_p)); \
+        (_iov_iter_p)->iov_offset = __src_iov_it; \
+        *(_dst_iov_cnt_p)         = __dst_iov_it; \
+        ((_max_length) - __remain_length); \
+    })
+
+
+/**
+ * Initializes the IOV iterator by the initial values.
+ *
+ * @param [in]     iov_iter        Pointer to the IOV iterator.
+ */
+static UCS_F_ALWAYS_INLINE
+void ucs_iov_iter_init(ucs_iov_iter_t *iov_iter)
+{
+    iov_iter->iov_offset    = 0;
+    iov_iter->buffer_offset = 0;
+}
+
+/**
+ * Sets the particular IOV data buffer.
+ *
+ * @param [in]     iov             Pointer to the IOV element.
+ * @param [in]     length          Length that needs to be set.
+ */
+static UCS_F_ALWAYS_INLINE
+void ucs_iov_set_length(struct iovec *iov, size_t length)
```

Comment:
pls rename all funcs which deal with "struct iovec" to "iovec":

ucs_iovec_set_length
ucs_iovec_set_buffer